source/org/zfin/marker/service/MarkerService.java

```diff
@@ -862,6 +862,28 @@ public static boolean markerHasAlias(Marker marker, String alias) {
         return false;
     }
 
+
+    public static MarkerAlias createMarkerAlias(Marker marker, String newAlias, List<String> publicationIDs) {
+        int loopIndex=0;
+        MarkerAlias alias = null;
+        for(String pubID : publicationIDs){
+            Publication publication = publicationRepository.getPublication(pubID);
+
+            // alias is created with the first reference. others added after the alias is created.
+            if (loopIndex == 0) {
+                alias = markerRepository.addMarkerAlias(marker, newAlias, publication);
+            }
+
+            // add direct attribution to aliased marker and provided publications
+            infrastructureRepository.insertStandardPubAttribution(marker.getZdbID(), publication);
+            markerRepository.addDataAliasAttribution(alias, publication, marker);
```

Comment:
Put this in the else of the "if (loopIndex == 0)" to make clear that it is not necessary for the first element.